scripts/quests/outlands/The_Missing_Piece.lua

```diff
@@ -0,0 +1,157 @@
+        end,
+
+        [xi.zone.RABAO] =
+        {
+            ['Alfesar'] =
+            {
+                onTrigger = function(player, npc)
+                    local progress = quest:getVar(player, 'Prog')
+
+                    if progress == 0 then
+                        return quest:event(7, xi.ki.ANCIENT_TABLET_FRAGMENT) -- Reminder to get KI
+                    elseif progress == 1 then
+                        return quest:progressEvent(8, xi.ki.ANCIENT_TABLET_FRAGMENT) -- Player has returned with KI
+                    elseif progress == 2 then
+                        return quest:event(9, 0, xi.ki.ANCIENT_TABLET_FRAGMENT) -- Reminder to go to Sandy to turn finish quest.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [8] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.ANCIENT_TABLET_FRAGMENT)
+                    player.addKeyItem(player, xi.ki.TABLET_OF_ANCIENT_MAGIC)
+                    player.addKeyItem(player, xi.ki.LETTER_FROM_ALFESAR)
+                    quest:setVar(player, 'Prog', 2)
+                    player:messageSpecial(rabaoID.text.ACCEPTED_KEYITEM, 0, xi.ki.TABLET_OF_ANCIENT_MAGIC, xi.ki.LETTER_FROM_ALFESAR)
+                end,
+            },
+        },
+
+        [xi.zone.QUICKSAND_CAVES] =
+        {
+            ['qm5'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        quest:setVar(player, 'Prog', 1)
+                        local newPosition = npcUtil.pickNewPosition(npc:getID(), positionTable)
+                        npc:setPos(newPosition.x, newPosition.y, newPosition.z)
+                        return quest:keyItem(xi.ki.ANCIENT_TABLET_FRAGMENT)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Charlaimagnat'] =
+            {
+                onTrigger = function(player, npc)
+                    local progress = quest:getVar(player, 'Prog')
+
+                    if progress == 2 then
+                        return quest:progressEvent(703)
+                    elseif
+                        progress == 3 and
+                        os.time() < player:getCharVar('TheMissingPiece_date')
+                    then
+                        return quest:event(704)
+                    elseif
+                        progress == 3 and
+                        os.time() >= player:getCharVar('TheMissingPiece_date')
+                    then
+                        return quest:progressEvent(705)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [703] = function(player, csid, option, npc)
+                    player:setCharVar('TheMissingPiece_date', os.time() + 60)
```

Comment:
could use quest:setVar(player, 'Wait', os.time() + 60) to keep in format and check same var up top